config/routes.rb

```diff
@@ -120,4 +55,7 @@
     end
   end
 
+  # get "*path" => redirect('/')
+  get '*a', :to => 'errors#routing'
+
```

Comment:
The *a is a wildcard that catches any undefined or misspelled route, then uses the routing method in  errors controller to redirect to the home page, and display a flash message saying the page does not exist.